### ISSUE TITLE
[Custom Weapons] Exhaustible weapons refer to temporary string for classname

### ISSUE THREAD

**Issue by fnky** [baso88/SC_AngelScript#27]
Exhaustible weapons ([`ItemInfo.iFlags`](https://baso88.github.io/SC_AngelScript/docs/ItemInfo.htm) & [`ITEM_FLAG_EXHAUSTIBLE`](https://baso88.github.io/SC_AngelScript/docs/ItemFlag.htm)) refer to a temporary string for the classname. This causes problems when you try to pick up dropped instances of these weapons. To fix this, go to `CASItemRegistry::RegisterItem`  and change the following line:

```cpp
UTIL_PrecacheOtherWeapon( szName.CStr() );
```

To:
```cpp
UTIL_PrecacheOtherWeapon( ALLOC_STRING( szName.CStr() ) );
```

This will ensure it uses a string that is valid for the duration of the map.

**Comment by fnky** [comment]
Trivial to fix, not in fixed in 5.05.

**Comment by baso88** [comment]
Fix for this issue will be part of the upcoming SC update (5.11).
